myenumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+      return to_a unless block_given?
+
+      yield self[index]
+      index += 1
+      self
+    end
+  end
+
+  def my_each_with_index(index = 0)
+    while index < size
+      return self unless block_given?
+
+      yield self[index], index
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
```

Comment:
* [ ]  Pass the name of the method as an argument to to_enum improve the customization
-  return to_enum(method name_name) unless block_given?